service/service_internal_test.go

```diff
@@ -0,0 +1,49 @@
+package service
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	pb "github.com/textileio/bidbot/gen/v1"
+	core "github.com/textileio/bidbot/lib/auction"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func TestClientFilter(t *testing.T) {
+	t.Parallel()
+
+	// Start with no client filters
+	s := Service{
+		auctionFilters: AuctionFilters{
+			DealDuration: MinMaxFilter{
+				Min: core.MinDealDuration,
+				Max: core.MaxDealDuration,
+			},
+			DealSize: MinMaxFilter{
+				Min: 1,
+				Max: 10,
+			},
+		},
+	}
+	auction := &pb.Auction{
+		DealSize:      5,
+		DealDuration:  core.MinDealDuration,
+		EndsAt:        timestamppb.New(time.Now().Add(time.Minute)),
+		ClientAddress: "f2kb4izxsxu2jyyslzwmv2sfbrgpld56efedgru5i",
+	}
+
+	t.Run("accept-any-client-address", func(t *testing.T) {
+		require.Empty(t, s.filterAuction(auction))
+	})
+
+	s.auctionFilters.ClientAddressWhitelist = []string{"f144zep4gitj73rrujd3jw6iprljicx6vl4wbeavi"}
+	t.Run("reject-single-client-address", func(t *testing.T) {
+		require.Contains(t, s.filterAuction(auction), "f2kb4izxsxu2jyyslzwmv2sfbrgpld56efedgru5i")
+	})
```

Comment:
Test with a whitelist only containing `f144zep4gitj73rrujd3jw6iprljicx6vl4wbeavi`.

Then test if an auction from client `f2kb4izxsxu2jyyslzwmv2sfbrgpld56efedgru5i` (L33) will be rejected, since that must be the case.

The rejection message should be a string explaining the cause of rejection, which must say something about `f2kb4izxsxu2jyyslzwmv2sfbrgpld56efedgru5i` being rejected from not being in the whitelist. The logic in the code never have a strict interpretation of the returned `string`, its only using for logging; so took the liberty of avoiding type-erroring.